app/src/main/java/org/schabi/newpipe/ui/components/items/stream/StreamMenu.kt

```diff
@@ -0,0 +1,152 @@
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.viewmodel.compose.viewModel
+import org.schabi.newpipe.R
+import org.schabi.newpipe.database.stream.model.StreamEntity
+import org.schabi.newpipe.download.DownloadDialog
+import org.schabi.newpipe.extractor.stream.StreamInfoItem
+import org.schabi.newpipe.local.dialog.PlaylistAppendDialog
+import org.schabi.newpipe.local.dialog.PlaylistDialog
+import org.schabi.newpipe.player.helper.PlayerHolder
+import org.schabi.newpipe.util.NavigationHelper
+import org.schabi.newpipe.util.SparseItemUtil
+import org.schabi.newpipe.util.external_communication.ShareUtils
+import org.schabi.newpipe.viewmodels.StreamViewModel
+
+@Composable
+fun StreamMenu(
+    stream: StreamInfoItem,
+    expanded: Boolean,
+    onDismissRequest: () -> Unit
+) {
+    val context = LocalContext.current
+    val streamViewModel = viewModel<StreamViewModel>()
+    val playerHolder = PlayerHolder.getInstance()
+
+    DropdownMenu(expanded = expanded, onDismissRequest = onDismissRequest) {
+        if (playerHolder.isPlayQueueReady) {
+            StreamMenuItem(
+                text = R.string.enqueue_stream,
+                onClick = {
+                    onDismissRequest()
+                    SparseItemUtil.fetchItemInfoIfSparse(context, stream) {
+                        NavigationHelper.enqueueOnPlayer(context, it)
+                    }
+                }
+            )
+
+            if (playerHolder.queuePosition < playerHolder.queueSize - 1) {
+                StreamMenuItem(
+                    text = R.string.enqueue_next_stream,
+                    onClick = {
+                        onDismissRequest()
+                        SparseItemUtil.fetchItemInfoIfSparse(context, stream) {
+                            NavigationHelper.enqueueNextOnPlayer(context, it)
+                        }
+                    }
+                )
+            }
+        }
+
+        StreamMenuItem(
+            text = R.string.start_here_on_background,
+            onClick = {
+                onDismissRequest()
+                SparseItemUtil.fetchItemInfoIfSparse(context, stream) {
+                    NavigationHelper.playOnBackgroundPlayer(context, it, true)
+                }
+            }
+        )
+        StreamMenuItem(
+            text = R.string.start_here_on_popup,
+            onClick = {
+                onDismissRequest()
+                SparseItemUtil.fetchItemInfoIfSparse(context, stream) {
+                    NavigationHelper.playOnPopupPlayer(context, it, true)
+                }
+            }
+        )
+        StreamMenuItem(
+            text = R.string.download,
+            onClick = {
+                onDismissRequest()
+                SparseItemUtil.fetchStreamInfoAndSaveToDatabase(
+                    context, stream.serviceId, stream.url
+                ) { info ->
+                    // TODO: Use an AlertDialog composable instead.
```

Comment:
See my comments above about "TODO" comments. In general, I thought it would make more sense to capture this in a new issue so that it can be prioritized and picked up by anyone interested in contributing to the code base. Please let me know your thoughts on this.

Reply:
I was planning to implement the download dialog in Compose in a separate PR.